_source/_posts/2022-10-20-spring-vault.md

```diff
@@ -0,0 +1,642 @@
+
+```shell
+SPRING_CLOUD_CONFIG_TOKEN=hvs.CAESIKd9pYyc9xesiqmwvep... \
+./mvnw spring-boot:run
+```
+
+When the `vault-demo-app` starts, it will request the configuration to the config server, which in turn will make REST requests to Vault. In the config server logs, with enough logging level, you will be able to see those requests:
+
+```shell
+2022-09-28 17:21:25.096 DEBUG 6685 --- [nio-8888-exec-1] o.s.web.client.RestTemplate              : HTTP GET http://127.0.0.1:8200/v1/secret/data/vault-demo-app,dev
+2022-09-28 17:21:25.142 DEBUG 6685 --- [nio-8888-exec-1] o.s.web.client.RestTemplate              : Accept=[application/json, application/*+json]
+2022-09-28 17:21:25.183 DEBUG 6685 --- [nio-8888-exec-1] o.s.web.client.RestTemplate              : Response 200 OK
+```
+
+Go to `http://localhost:8080` and verify that authentication with Okta works.
+
+Finally, let's seal Vault. Sealing allows you to lock Vault data to minimize damages when an intrusion is detected.  In the container command line, enter:
+
+```shell
+vault operator seal
+```
+
+Restart `vault-demo-app` and verify the configuration will not be retrieved as Vault is sealed. The `vault-config-server` logs should read:
+
+```
+503 Service Unavailable: "{"errors":["Vault is sealed"]}<EOL>"]
+```
+
+To unseal Vault, run:
+
+```shell
+vault operator unseal {unsealKey}
+```
+
+## Learn More About Encryption and Storing Secrets
+
+Hopefully you see the benefits of using a secrets management tool like Vault as a configuration backend, as opposed to storing secrets in a file, on a file system, or in a code repository. To learn more about Vault and Spring Cloud, check out the following links:
+
+* [How Bad Can It Git?](https://www.ndss-symposium.org/ndss-paper/how-bad-can-it-git-characterizing-secret-leakage-in-public-github-repositories/)
+* [Spring Cloud Config - Vault Backend](https://cloud.spring.io/spring-cloud-config/reference/html/#vault-backend)
+* [Container Secrets with Vault](https://www.hashicorp.com/resources/securing-container-secrets-vault)
+* [Amazon Engineer Leaked Private Encryption Keys](https://gizmodo.com/amazon-engineer-leaked-private-encryption-keys-outside-1841160934)
+
+We have several related posts about encryption and storing secrets on this blog.
+
+* [Five Anti-Patterns with Secrets in Java](/blog/2021/12/14/antipatterns-secrets-java)
+* [Security Patterns for Microservice Architectures](/blog/2020/03/23/microservice-security-patterns)
+* [How to Secure Your Kubernetes Clusters With Best Practices](/blog/2021/12/02/k8s-security-best-practices)
+
+You can find the code for this tutorial on GitHub in the [@oktadev/okta-spring-vault-example](https://github.com/oktadev/okta-spring-vault-example) repository.
```

Comment:
I created a PR from [your spring-vault-2.7.3 repo](https://github.com/indiepopart/spring-vault-2.7.3) to https://github.com/oktadev/okta-spring-vault-example:

https://github.com/oktadev/okta-spring-vault-example/pull/2

Reply:
I see the PR is for merging from okta-spring-vault 2.7.4 branch to okta-spring-vault main. I updated Vault version in my repo. Please let me know if should update somewhere else.

Reply:
As long as https://github.com/oktadev/okta-spring-vault-example/pull/2 matches your repo, I think everything is good to go.

Reply:
I sent a pr for the vault version update https://github.com/oktadev/okta-spring-vault-example/pull/3